drv/cpu-seq-api/src/lib.rs

```diff
@@ -31,6 +32,22 @@ pub enum SeqError {
     ServerRestarted,
 }
 
+#[derive(Copy, Clone, Debug, FromPrimitive, Eq, PartialEq, AsBytes, Count)]
+#[repr(u8)]
+pub enum StateChangeReason {
+    /// The system has just received power, so the sequencer has booted the
+    /// host CPU.
+    InitialPowerOn = 1,
+    /// A power state change was requested by the control plane.
+    ControlPlane,
+    /// The host OS requested that the system power off without rebooting.
+    HostPowerOff,
+    /// The host OS requested that the system reboot.
+    HostReboot,
```

Comment:
While I realize it's a bit tricky, I think it'd be useful to distinguish between a reboot due to an IPCC panic and a reboot. Unfortunately the panic command doesn't itself indicate a reboot, but a subsequent one from the host with no additional action taken will indicate a panic. It may not be relevant at this particular point, but it depends a bit on where we see us generating an alert that the system has rebooted from. That is, it's not clear to me that we'd want to do that from all of the different callers versus use the sequencer as a generator.

I also somewhat expected to see a distinction between overheat and MAPO related actions, but the latter may be because if the FPGA sequencer triggers a MAPO event we are seeing that in a different way. 

If someone uses humility to manually turn on/off the host power how is that noted through here?

Reply:
> While I realize it's a bit tricky, I think it'd be useful to distinguish between a reboot due to an IPCC panic and a reboot. Unfortunately the panic command doesn't itself indicate a reboot, but a subsequent one from the host with no additional action taken will indicate a panic.

Definitely agree that it would be good to differentiate between these, I'll see about having the host-sp-comms task track a "was there just a panic" bit so we can thread that through here.

> I also somewhat expected to see a distinction between overheat and MAPO related actions, but the latter may be because if the FPGA sequencer triggers a MAPO event we are seeing that in a different way.

As I understand it, MAPO events won't go through the `Sequencer.set_state` IPC interface, and won't record the `SetState` ringbuf entry. Instead, I think Hubris' understanding of MAPO is entirely internal to the sequencer task --- I think the relevant ringbuf entries are the ones we record in [`a0_failure` here](https://github.com/oxidecomputer/hubris/blob/63995e0cdd0e5087244bbd413db7d00b6654bfc5/drv/gimlet-seq-server/src/main.rs#L937-L949). @bcantrill or @cbiffle would know better, but that's my understanding, at least.

Reply:
We're now able to distinguish between host boot failures, host panics, and host reboot/power-off requests, as of https://github.com/oxidecomputer/hubris/pull/1953/commits/2650ced4a3bcfaf0ef1384ee64b1f0961e794e2b. The change I made in https://github.com/oxidecomputer/hubris/pull/1953/commits/7e6b0d31f80a9821e6a72ec22821b8fe6611cbb3 to leave the IPCs currently used by Hiffy alone to avoid breaking scripts etc should cause any Hiffy-triggered power state change to have the reason `Other`. I considered explicitly making it `Hiffy`, but :woman_shrugging: